CollectorUtils/reset_required_fields.py

```diff
@@ -0,0 +1,153 @@
+    :param expiration: How long to make the token valid for (minutes)
+    :return: the token as a string
+    """
+    url = urlparse.urljoin(org_url, "sharing/rest/generateToken")
+    data = dict(username=username, password=password, referer=org_url, f='json', expiration=expiration)
+    response = post(url, data)
+    token = response["token"]
+    return token
+
+
+def update_required_fields(item_id, org_url, token):
+    """
+    This sets the 'required fields' to in the template to be null
+    rather than empty white spaces or 0. Required fields are fields that
+    have the nullable property set to false
+    :param item_id: The id of the service to update
+    :param org_url: The url of the ogranization
+    :param token: The token of an authenticated user
+    :return: List of feature service layer urls that weren't updated successfully
+    """
+    # set the item url
+    errors = []
+    url = "{}/sharing/rest/content/items/{}".format(org_url, item_id)
+    params = {'token': token}
+    # get the items service url
+    service_url = get(url, params)["url"]
+    # get all of the layer ids for the service
+    layer_ids = [layer["id"] for layer in get(service_url, params)["layers"]]
+    # for each layer
+    for layer_id in layer_ids:
+        fields_not_null = []
+        layer_url = "{}/{}".format(service_url, layer_id)
+        # inject 'admin' into url so we can hit the updateDefinition endpoint
+        index = layer_url.index("/services/")
+        admin_layer_url = layer_url[0:index] + "/admin" + layer_url[index:]
+        layer_dict = get(admin_layer_url, params)
+        if "fields" in layer_dict:
+            for field in layer_dict["fields"]:
+                if not field["nullable"]:
+                    fields_not_null.append(field["name"])
+            if "templates" in layer_dict:
+                for template in layer_dict["templates"]:
+                    if "prototype" in template and "attributes" in template["prototype"]:
+                        for field_name, value in template["prototype"]["attributes"].items():
+                            if field_name in fields_not_null:
+                                template["prototype"]["attributes"][field_name] = None
+        # Need to remove lastEditDate so that the definition is actually updated
+        if "editingInfo" in layer_dict and "lastEditDate" in layer_dict["editingInfo"]:
+            del layer_dict["editingInfo"]["lastEditDate"]
+        update_params = dict(token=token, f="json",
+                             updateDefinition=json.dumps(layer_dict).encode('utf8', errors='replace'))
+        # hit the updateDefinition end point to update the definition
+        res = post("{}/updateDefinition".format(admin_layer_url), update_params)
+        arcpy.AddError(res)
+        if "success" not in res:
+            errors.append(layer_url)
+    return errors
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser("Configure GNSS Popup Info")
```

Comment:
Can you change the strings here to say "Reset Required fields to null" ... my bad for the copy-paste error